v2/apiserver/internal/core/containers.go

```diff
@@ -23,3 +23,50 @@ type ContainerSpec struct {
 	// to be set within the OCI container.
 	Environment map[string]string `json:"environment,omitempty" bson:"environment,omitempty"` // nolint: lll
 }
+
+// DeepEquals returns a boolean value indicating whether or not a ContainerSpec
+// is equivalent to another.
+func (c ContainerSpec) DeepEquals(c2 ContainerSpec) bool {
```

Comment:
I'm worried that we'll overlook updating this if we ever add a field.

I'm generally opposed to using reflection in production code because it's slow, but I wonder if `reflect.DeepEqual()` may be our best option for this.

Reply:
Good concern.  Ok, updated to use reflect.